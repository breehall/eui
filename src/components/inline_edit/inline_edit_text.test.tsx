/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { render } from '../../test/rtl';
import { requiredProps } from '../../test/required_props';

import { EuiInlineEditText, EuiInlineEditTextProps } from './inline_edit_text';
import { TEXT_SIZES } from '../text/text';

describe('EuiInlineEditTitle', () => {
  test('renders as title in readMode', () => {
    const { container } = render(
      <EuiInlineEditText
        inputAriaLabel={'textInputReadModeTest'}
        defaultValue="hello world"
        {...requiredProps}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders as title in editMode', () => {
    const { container } = render(
      <EuiInlineEditText
        inputAriaLabel={'textInputEditModeTest'}
        defaultValue="hello world"
        startWithEditOpen={true}
        {...requiredProps}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  describe('props', () => {
    describe('sizes', () => {
      // Remove 'relative' from text sizes available for EuiInlineEditText
      const usedTextSizes = TEXT_SIZES.filter((size) => size !== 'relative');

      usedTextSizes.forEach((size: string) => {
        test(`${size} is rendered in readMode`, () => {
          const component = render(
            <EuiInlineEditText
              size={size as EuiInlineEditTextProps['size']}
              defaultValue="Hello World!"
              inputAriaLabel="inlineEditLabel"
              saveButtonAriaLabel="saveEditButton"
            />
          );

          expect(component).toMatchSnapshot();
        });
      });

      usedTextSizes.forEach((size: string) => {
        test(`${size} is rendered in editMode`, () => {
          const component = render(
            <EuiInlineEditText
              size={size as EuiInlineEditTextProps['size']}
              defaultValue="Hello World!"
              inputAriaLabel="inlineEditLabel"
              saveButtonAriaLabel="saveEditButton"
              startWithEditOpen={true}
            />
          );

          expect(component).toMatchSnapshot();
        });
      });
    });
  });
});
