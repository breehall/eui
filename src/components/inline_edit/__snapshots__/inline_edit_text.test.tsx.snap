// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiInlineEditTitle props sizes m is rendered in editMode 1`] = `
<div
  class="euiInlineEdit euiInlineEditText emotion-euiInlineEditText-m-m"
>
  <div
    class="euiForm"
  >
    <div
      class="euiFlexGroup emotion-euiFlexGroup-responsive-s-flexStart-stretch-row"
    >
      <div
        class="euiFlexItem emotion-euiFlexItem-grow-1"
      >
        <div
          class="euiFormRow euiFormRow--fullWidth"
          id="generated-id-row"
        >
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiFormControlLayout euiFormControlLayout--fullWidth"
            >
              <div
                class="euiFormControlLayout__childrenWrapper"
              >
                <input
                  aria-invalid="false"
                  aria-label="inlineEditLabel"
                  class="euiFieldText euiFieldText--fullWidth"
                  id="generated-id"
                  type="text"
                  value="Hello World!"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem euiInlineEdit euiInlineEditText emotion-euiFlexItem-growZero-euiInlineEditText-m-m"
      >
        <div
          class="euiFormRow euiFormRow--fullWidth"
          id="generated-id-row"
        >
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              aria-busy="false"
              data-test-subj="euiSkeletonLoadingAriaWrapper"
            >
              <div
                class="emotion-euiScreenReaderOnly"
              >
                <div
                  aria-atomic="true"
                  aria-live="polite"
                  role="status"
                >
                  Loaded 
                </div>
                <div
                  aria-atomic="true"
                  aria-hidden="true"
                  aria-live="off"
                  role="status"
                />
              </div>
              <button
                aria-label="saveEditButton"
                class="euiButtonIcon euiButtonIcon--medium emotion-euiButtonIcon-base-success"
                type="button"
              >
                <span
                  aria-hidden="true"
                  class="euiButtonIcon__icon"
                  color="inherit"
                  data-euiicon-type="check"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem emotion-euiFlexItem-growZero"
      >
        <div
          class="euiFormRow euiFormRow--fullWidth"
          id="generated-id-row"
        >
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              aria-busy="false"
              data-test-subj="euiSkeletonLoadingAriaWrapper"
            >
              <div
                class="emotion-euiScreenReaderOnly"
              >
                <div
                  aria-atomic="true"
                  aria-live="polite"
                  role="status"
                >
                  Loaded 
                </div>
                <div
                  aria-atomic="true"
                  aria-hidden="true"
                  aria-live="off"
                  role="status"
                />
              </div>
              <button
                aria-label="Cancel edit"
                class="euiButtonIcon euiButtonIcon--medium emotion-euiButtonIcon-base-danger"
                type="button"
              >
                <span
                  aria-hidden="true"
                  class="euiButtonIcon__icon"
                  color="inherit"
                  data-euiicon-type="cross"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EuiInlineEditTitle props sizes m is rendered in readMode 1`] = `
<div
  class="euiInlineEdit euiInlineEditText emotion-euiInlineEditText-m-m"
>
  <button
    class="euiButtonEmpty euiButtonEmpty--flushBoth css-wvaqcf-empty-text"
    type="button"
  >
    <span
      class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
    >
      <span
        class="euiButtonContent__icon"
        color="inherit"
        data-euiicon-type="pencil"
      />
      <span
        class="euiButtonEmpty__text"
      >
        <div
          class="euiText eui-textTruncate emotion-euiText-m"
        >
          Hello World!
        </div>
      </span>
    </span>
  </button>
</div>
`;

exports[`EuiInlineEditTitle props sizes s is rendered in editMode 1`] = `
<div
  class="euiInlineEdit euiInlineEditText emotion-euiInlineEditText-s-s"
>
  <div
    class="euiForm"
  >
    <div
      class="euiFlexGroup emotion-euiFlexGroup-responsive-s-flexStart-stretch-row"
    >
      <div
        class="euiFlexItem emotion-euiFlexItem-grow-1"
      >
        <div
          class="euiFormRow euiFormRow--fullWidth"
          id="generated-id-row"
        >
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiFormControlLayout euiFormControlLayout--fullWidth euiFormControlLayout--compressed"
            >
              <div
                class="euiFormControlLayout__childrenWrapper"
              >
                <input
                  aria-invalid="false"
                  aria-label="inlineEditLabel"
                  class="euiFieldText euiFieldText--fullWidth euiFieldText--compressed"
                  id="generated-id"
                  type="text"
                  value="Hello World!"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem euiInlineEdit euiInlineEditText emotion-euiFlexItem-growZero-euiInlineEditText-s-s"
      >
        <div
          class="euiFormRow euiFormRow--fullWidth"
          id="generated-id-row"
        >
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              aria-busy="false"
              data-test-subj="euiSkeletonLoadingAriaWrapper"
            >
              <div
                class="emotion-euiScreenReaderOnly"
              >
                <div
                  aria-atomic="true"
                  aria-live="polite"
                  role="status"
                >
                  Loaded 
                </div>
                <div
                  aria-atomic="true"
                  aria-hidden="true"
                  aria-live="off"
                  role="status"
                />
              </div>
              <button
                aria-label="saveEditButton"
                class="euiButtonIcon euiButtonIcon--small emotion-euiButtonIcon-base-success"
                type="button"
              >
                <span
                  aria-hidden="true"
                  class="euiButtonIcon__icon"
                  color="inherit"
                  data-euiicon-type="check"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem emotion-euiFlexItem-growZero"
      >
        <div
          class="euiFormRow euiFormRow--fullWidth"
          id="generated-id-row"
        >
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              aria-busy="false"
              data-test-subj="euiSkeletonLoadingAriaWrapper"
            >
              <div
                class="emotion-euiScreenReaderOnly"
              >
                <div
                  aria-atomic="true"
                  aria-live="polite"
                  role="status"
                >
                  Loaded 
                </div>
                <div
                  aria-atomic="true"
                  aria-hidden="true"
                  aria-live="off"
                  role="status"
                />
              </div>
              <button
                aria-label="Cancel edit"
                class="euiButtonIcon euiButtonIcon--small emotion-euiButtonIcon-base-danger"
                type="button"
              >
                <span
                  aria-hidden="true"
                  class="euiButtonIcon__icon"
                  color="inherit"
                  data-euiicon-type="cross"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EuiInlineEditTitle props sizes s is rendered in readMode 1`] = `
<div
  class="euiInlineEdit euiInlineEditText emotion-euiInlineEditText-s-s"
>
  <button
    class="euiButtonEmpty euiButtonEmpty--small euiButtonEmpty--flushBoth css-wvaqcf-empty-text"
    type="button"
  >
    <span
      class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
    >
      <span
        class="euiButtonContent__icon"
        color="inherit"
        data-euiicon-type="pencil"
      />
      <span
        class="euiButtonEmpty__text"
      >
        <div
          class="euiText eui-textTruncate emotion-euiText-s"
        >
          Hello World!
        </div>
      </span>
    </span>
  </button>
</div>
`;

exports[`EuiInlineEditTitle props sizes xs is rendered in editMode 1`] = `
<div
  class="euiInlineEdit euiInlineEditText emotion-euiInlineEditText-xs-xs"
>
  <div
    class="euiForm"
  >
    <div
      class="euiFlexGroup emotion-euiFlexGroup-responsive-s-flexStart-stretch-row"
    >
      <div
        class="euiFlexItem emotion-euiFlexItem-grow-1"
      >
        <div
          class="euiFormRow euiFormRow--fullWidth"
          id="generated-id-row"
        >
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiFormControlLayout euiFormControlLayout--fullWidth euiFormControlLayout--compressed"
            >
              <div
                class="euiFormControlLayout__childrenWrapper"
              >
                <input
                  aria-invalid="false"
                  aria-label="inlineEditLabel"
                  class="euiFieldText euiFieldText--fullWidth euiFieldText--compressed"
                  id="generated-id"
                  type="text"
                  value="Hello World!"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem euiInlineEdit euiInlineEditText emotion-euiFlexItem-growZero-euiInlineEditText-xs-xs"
      >
        <div
          class="euiFormRow euiFormRow--fullWidth"
          id="generated-id-row"
        >
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              aria-busy="false"
              data-test-subj="euiSkeletonLoadingAriaWrapper"
            >
              <div
                class="emotion-euiScreenReaderOnly"
              >
                <div
                  aria-atomic="true"
                  aria-live="polite"
                  role="status"
                >
                  Loaded 
                </div>
                <div
                  aria-atomic="true"
                  aria-hidden="true"
                  aria-live="off"
                  role="status"
                />
              </div>
              <button
                aria-label="saveEditButton"
                class="euiButtonIcon euiButtonIcon--small emotion-euiButtonIcon-base-success"
                type="button"
              >
                <span
                  aria-hidden="true"
                  class="euiButtonIcon__icon"
                  color="inherit"
                  data-euiicon-type="check"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem emotion-euiFlexItem-growZero"
      >
        <div
          class="euiFormRow euiFormRow--fullWidth"
          id="generated-id-row"
        >
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              aria-busy="false"
              data-test-subj="euiSkeletonLoadingAriaWrapper"
            >
              <div
                class="emotion-euiScreenReaderOnly"
              >
                <div
                  aria-atomic="true"
                  aria-live="polite"
                  role="status"
                >
                  Loaded 
                </div>
                <div
                  aria-atomic="true"
                  aria-hidden="true"
                  aria-live="off"
                  role="status"
                />
              </div>
              <button
                aria-label="Cancel edit"
                class="euiButtonIcon euiButtonIcon--small emotion-euiButtonIcon-base-danger"
                type="button"
              >
                <span
                  aria-hidden="true"
                  class="euiButtonIcon__icon"
                  color="inherit"
                  data-euiicon-type="cross"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EuiInlineEditTitle props sizes xs is rendered in readMode 1`] = `
<div
  class="euiInlineEdit euiInlineEditText emotion-euiInlineEditText-xs-xs"
>
  <button
    class="euiButtonEmpty euiButtonEmpty--small euiButtonEmpty--flushBoth css-wvaqcf-empty-text"
    type="button"
  >
    <span
      class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
    >
      <span
        class="euiButtonContent__icon"
        color="inherit"
        data-euiicon-type="pencil"
      />
      <span
        class="euiButtonEmpty__text"
      >
        <div
          class="euiText eui-textTruncate emotion-euiText-xs"
        >
          Hello World!
        </div>
      </span>
    </span>
  </button>
</div>
`;

exports[`EuiInlineEditTitle renders as title in editMode 1`] = `
<div
  class="euiInlineEdit euiInlineEditText testClass1 testClass2 emotion-euiInlineEditText-m-m"
>
  <div
    class="euiForm"
  >
    <div
      class="euiFlexGroup emotion-euiFlexGroup-responsive-s-flexStart-stretch-row"
    >
      <div
        class="euiFlexItem emotion-euiFlexItem-grow-1"
      >
        <div
          class="euiFormRow euiFormRow--fullWidth"
          id="generated-id-row"
        >
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiFormControlLayout euiFormControlLayout--fullWidth"
            >
              <div
                class="euiFormControlLayout__childrenWrapper"
              >
                <input
                  aria-invalid="false"
                  aria-label="textInputEditModeTest"
                  class="euiFieldText euiFieldText--fullWidth"
                  id="generated-id"
                  type="text"
                  value="hello world"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem euiInlineEdit euiInlineEditText testClass1 testClass2 emotion-euiFlexItem-growZero-euiInlineEditText-m-m"
      >
        <div
          class="euiFormRow euiFormRow--fullWidth"
          id="generated-id-row"
        >
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              aria-busy="false"
              data-test-subj="euiSkeletonLoadingAriaWrapper"
            >
              <div
                class="emotion-euiScreenReaderOnly"
              >
                <div
                  aria-atomic="true"
                  aria-live="polite"
                  role="status"
                >
                  Loaded 
                </div>
                <div
                  aria-atomic="true"
                  aria-hidden="true"
                  aria-live="off"
                  role="status"
                />
              </div>
              <button
                aria-label="Save edit"
                class="euiButtonIcon euiButtonIcon--medium emotion-euiButtonIcon-base-success"
                type="button"
              >
                <span
                  aria-hidden="true"
                  class="euiButtonIcon__icon"
                  color="inherit"
                  data-euiicon-type="check"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem emotion-euiFlexItem-growZero"
      >
        <div
          class="euiFormRow euiFormRow--fullWidth"
          id="generated-id-row"
        >
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              aria-busy="false"
              data-test-subj="euiSkeletonLoadingAriaWrapper"
            >
              <div
                class="emotion-euiScreenReaderOnly"
              >
                <div
                  aria-atomic="true"
                  aria-live="polite"
                  role="status"
                >
                  Loaded 
                </div>
                <div
                  aria-atomic="true"
                  aria-hidden="true"
                  aria-live="off"
                  role="status"
                />
              </div>
              <button
                aria-label="Cancel edit"
                class="euiButtonIcon euiButtonIcon--medium emotion-euiButtonIcon-base-danger"
                type="button"
              >
                <span
                  aria-hidden="true"
                  class="euiButtonIcon__icon"
                  color="inherit"
                  data-euiicon-type="cross"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EuiInlineEditTitle renders as title in readMode 1`] = `
<div
  class="euiInlineEdit euiInlineEditText testClass1 testClass2 emotion-euiInlineEditText-m-m"
>
  <button
    class="euiButtonEmpty euiButtonEmpty--flushBoth css-wvaqcf-empty-text"
    type="button"
  >
    <span
      class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
    >
      <span
        class="euiButtonContent__icon"
        color="inherit"
        data-euiicon-type="pencil"
      />
      <span
        class="euiButtonEmpty__text"
      >
        <div
          class="euiText eui-textTruncate emotion-euiText-m"
        >
          hello world
        </div>
      </span>
    </span>
  </button>
</div>
`;
